docker/docker-compose.yml

```
@@ -0,0 +1,106 @@
+volumes:
+  bitcoin-data:
+  bitcoin-bin:
+
+
+services:
+  bitcoin-node:
+    image: bitcoin-node
+    build:
+      context: .
+      dockerfile: bitcoin-node.dockerfile
+    command:
+      ["/bin/sh", "-c", "/shared/bitcoind -regtest -printtoconsole"]
+    ports:
+      - "8332:8332"
+      - "8333:8333"
+    volumes:
+      - bitcoin-data:/home/bitcoin/.bitcoin
+      - bitcoin-bin:/shared
+    healthcheck:
+      test: ["CMD", "bitcoin-cli", "-regtest", "getblockchaininfo"]
+      interval: 1m
+      timeout: 3s
+      retries: 3
+      start_period: 5s
+
+  nats:
+    image: nats
+    build:
+      context: .
+      dockerfile: nats.dockerfile
+    ports:
+      - "4222:4222"
+    healthcheck:
+      test: ["CMD-SHELL", "curl -sf http://localhost:8222/healthz || exit 1"]
+      interval: 1m
+      timeout: 1s
+      retries: 3
+      start_period: 1s
+
+  peer-extractor:
+    image: peer-base
+    build:
+      context: .
+      dockerfile: peer-base.dockerfile
+    command:
+      ["sudo", "-E", "/usr/local/cargo/bin/cargo", "run", "--bin", "extractor", "--", "--nats-address", "nats://nats:4222", "--bitcoind-path", "/shared/bitcoind"]
```

you probably want to have the release builds here as well
```suggestion
      ["sudo", "-E", "/usr/local/cargo/bin/cargo", "run", "--release", "--bin", "extractor", "--", "--nats-address", "nats://nats:4222", "--bitcoind-path", "/shared/bitcoind"]
```